internal/provider/kubernetes/routes.go

```diff
@@ -152,6 +151,17 @@ func (r *gatewayAPIReconciler) processGRPCRoutes(ctx context.Context, gatewayNam
 func (r *gatewayAPIReconciler) processHTTPRoutes(ctx context.Context, gatewayNamespaceName string,
 	resourceMap *resourceMappings, resourceTree *gatewayapi.Resources) error {
 	httpRouteList := &gwapiv1b1.HTTPRouteList{}
+
+	// An HTTPRoute may reference an AuthenticationFilter, so add them to the resource map first (if they exist).
+	filters, err := r.getAuthenticationFilters(ctx)
+	if err != nil {
+		return err
+	}
+	for i := range filters {
+		filter := filters[i]
```

Comment:
curious why this step is needed

Reply:
Do to implicit memory aliasing in for loop:
```
$ make lint
===========> Running go.mod.lint ...
===========> Go module looks clean!
===========> Running lint.golint ...
tools/bin/golangci-lint run  --build-tags=e2e --config=tools/linter/golangci-lint/.golangci.yml
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
internal/provider/kubernetes/routes.go:161:50: G601: Implicit memory aliasing in for loop. (gosec)
		resourceMap.authenFilters[utils.NamespacedName(&filter)] = &filter
		                                               ^
make[1]: *** [lint.golint] Error 1
```